lib/internal/Magento/Framework/App/Bootstrap.php

```diff
@@ -430,6 +430,10 @@ protected function terminate(\Throwable $e)
     {
 
         if ($this->isDeveloperMode()) {
```

Comment:
Can we add single such block here and remove it in those 2 places below?
```suggestion
        if (!headers_sent()) {
            header('HTTP/1.1 500 Internal Server Error');
        }

        if ($this->isDeveloperMode()) {
```